cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesExport.java

```diff
@@ -67,20 +68,49 @@ public void export(Network network, Properties params, DataSource ds, Reporter r
         String filenameTp = baseName + "_TP.xml";
         String filenameSsh = baseName + "_SSH.xml";
         String filenameSv = baseName + "_SV.xml";
+
+        // Reference data (if required) will come from imported boundaries
+        // We may have received a sourcing actor as a parameter
+        String sourcingActorName = Parameter.readString(getFormat(), params, SOURCING_ACTOR_PARAMETER, defaultValueConfig);
+        String countryName = null;
+        if (sourcingActorName == null || sourcingActorName.isEmpty()) {
```

Comment:
I disagree with that logic. For me, it is better to rely on the country present in the file. If you find one country and with the boundary files you are able to find the information about geographical region, that is nice. If you are not able, because the country leads to several sourcing actors for example and you are not able to choose, so you use the parameter. But for me, using the parameter is only a back up. Maybe I was not clear on that sorry.